File: src/main/webapp/src/app/models/action-lighting.ts
import { Action } from "./action";
import { LightingActionUniverse } from "./lighting-action-universe";

export class ActionLighting extends Action {
  lightingActionUniverseList: LightingActionUniverse[] = [];

  constructor(data?: any) {
    super(data);

    if (!data) {
      return;
    }

    if (data.lightingActionUniverseList) {
      for (let lightingActionUniverse of data.lightingActionUniverseList) {
        this.lightingActionUniverseList.push(
          new LightingActionUniverse(lightingActionUniverse)
        );
      }
    }

    if (this.lightingActionUniverseList.length === 0) {
      this.lightingActionUniverseList.push(new LightingActionUniverse());
    }
  }

  toJSON() {
    return { actionLighting: { ...this } };
  }
}

File: src/main/java/com/ascargon/rocketshow/lighting/ActionLighting.java
package com.ascargon.rocketshow.lighting;

import com.ascargon.rocketshow.util.Action;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class ActionLighting extends Action {

    private List<LightingActionUniverse> lightingActionUniverseList = new ArrayList<>();

    @Override
    public ActionType getType() {
        return ActionType.LIGHTING;
    }
}
